### ISSUE TITLE
Update Home page to include main components and their GitHub Star count

### ISSUE THREAD

**Issue by paravatha** [kubeflow/website#3624]
**Feature request**
On the https://www.kubeflow.org/ home page, add the below components **image/logo** with their **Github repo** link and the respective **Github repo star count** 

- Kubeflow, https://github.com/kubeflow/kubeflow, 13.1k stars
- Pipelines, https://github.com/kubeflow/pipelines, 3.4k stars
- KServe, https://github.com/kserve/kserve, 2.6k stars
- Katib, https://github.com/kubeflow/katib, 1.4k stars 
- Training Operator, https://github.com/kubeflow/training-operator, 1.4k stars

Something like this is done by Argo project for their components, reference https://argoproj.github.io/

**Comment by varodrig** [comment]
I'm not sure about this to be honest. 
any thoughts from the community?
cc [@kubeflow/kubeflow-steering-committee](https://github.com/orgs/kubeflow/teams/kubeflow-steering-committee) [@kubeflow/wg-training-leads](https://github.com/orgs/kubeflow/teams/wg-training-leads) [@kubeflow/wg-pipeline-leads](https://github.com/orgs/kubeflow/teams/wg-pipeline-leads) [@kubeflow/wg-notebooks-leads](https://github.com/orgs/kubeflow/teams/wg-notebooks-leads) [@kubeflow/wg-manifests-leads](https://github.com/orgs/kubeflow/teams/wg-manifests-leads) [@kubeflow/release-team](https://github.com/orgs/kubeflow/teams/release-team) [@kubeflow/wg-data-leads](https://github.com/orgs/kubeflow/teams/wg-data-leads) [@kubeflow/wg-deployment-leads](https://github.com/orgs/kubeflow/teams/wg-deployment-leads)

**Comment by jaiakash** [comment]
Hi, I feel at least for KubeFlow we should have github star banner for sure. Something like this, I will improve the UI and attract new contributors.

![Image](https://github.com/user-attachments/assets/a1efe747-463d-4d1a-96dc-1293b075c9c1)

But for individual components to have that banner, I am personally not very fan of that since these components come under the umbrella of Kubeflow MLF, while Argo's individual tools are a bit more independent from each other.

Feel free to suggest what you all think would be better.

**Comment by jaiakash** [comment]
Your thoughts on the above suggestion, @varodrig @andreyvelich ?

**Comment by andreyvelich** [comment]
I would prefer if we have star banner for every Kubeflow project, since all of them can be used as a standalone project.
Maybe we can add it to the homepage: https://www.kubeflow.org/

**Comment by andreyvelich** [comment]
Looks amazing @jaiakash.
WDYT @thesuperzapper ?

**Comment by jaiakash** [comment]
<img width="457" alt="Image" src="https://github.com/user-attachments/assets/3d393881-6c42-4a85-ae85-afcbedda0fd4" />

For the main kubeflow repo, I added it in the navbar. Personally I feel for this navbar we should have a better design.

Fork link: https://github.com/jaiakash/website/tree/github-stars

**Comment by thesuperzapper** [comment]
@andreyvelich I am ok with having the `kubeflow/kubeflow` stars at the top, but I think we need to format it a bit more consistently with the page, especially considering:

1. What it looks like on mobile (or with small horizontal screen)
2. What it might look like on the new dark mode (https://github.com/kubeflow/website/pull/3981)

---

However, I don't like the per-component start counts on the homepage for a few reasons:

1. it distracts from the card titles
2. it creates a bunch of new clickable links on the homepage
3. it is less clear what the intended "action" from these cards is
      - _we should be trying to have people click through to the docs for the components they are interested in, not leave the website to GitHub_

Instead, we could possibly include the star counts on the overview page on each component, instead of the homepage.

**Comment by thesuperzapper** [comment]
Also, there is a slight privacy (and website-reliability) concern with including the "shields" image (which is hosted by another service) on literally every page of the website.

**Comment by terrytangyuan** [comment]
I have a slightly different opinion here. GitHub stars is a misleading metric and people often take it in the wrong way. Not a strong objection though. 

If we do end up doing this, we should not include stars from `kubeflow/kubeflow` repo.

**Comment by jaiakash** [comment]
I agree with @thesuperzapper 's first point, We should have a star count for `kubeflow/kubeflow` on the main page while considering mobile and dark modes. But seeing https://github.com/kubeflow/kubeflow/issues/7549, that option is sustainable only till the `kubeflow/kubeflow` is active.

For ques, 
> 2. it creates a bunch of new clickable links on the homepage
- Nope, those Github start components are not clickable in themselves.
> 3. it is less clear what the intended "action" from these cards is. we should be trying to have people click through to the docs for the components they are interested in, not leave the website to GitHub
-  Yeah, that is what is happening; if the user clicks on that, it redirects to the doc page of that component, not Github.


-------------------------------
> 1. it distracts from the card titles
> Instead, we could include the star counts on the overview page on each component instead of the homepage.

Actually, I am looking to create a custom component that is more suited for the KubeFlow website's theme, covering both mobile view and dark mode. 
But my doubt is how to add it to our current project, which is a hugo server serving static files.

**Comment by franciscojavierarceo** [comment]
I agree with @terrytangyuan , we probably shouldn't include stars from kubeflow/kubeflow. Otherwise, I'm good with the inclusion.

I do think having the links clickable to enable users to star the repo is useful but I understand that may not be ideal behavior.

**Comment by varodrig** [comment]
> I have a slightly different opinion here. GitHub stars is a misleading metric and people often take it in the wrong way. Not a strong objection though.
> 
> If we do end up doing this, we should not include stars from `kubeflow/kubeflow` repo.

Same thoughts as @terrytangyuan

**Comment by eliterks** [comment]
@varodrig is the issue still active? if yes , could you please assign the issue to me?

**Comment by adityasah104** [comment]
Hello everyone! Is this issue still active? I can work on it as I am new into contributing and looking for good first issue.

**Comment by adityasah104** [comment]
Hi team! Based on the discussion, here's my understanding:

**Proposed Implementation:**
- Add GitHub stars for: Pipelines, KServe, Katib, Training Operator
- Exclude: kubeflow/kubeflow repo
- Location: Component overview pages (not main homepage)
- Design: Custom component, no shields.io, mobile + dark mode compatible, clickable to GitHub

**Need confirmation on:**
1. Is component overview pages (vs homepage) the agreed location? (@thesuperzapper suggested this, but @andreyvelich mentioned homepage)
2. Should stars be fetched at build-time or dynamically?
3. Any specific design reference besides Argo CD?

I'll create mockups once we align on these points.

cc @andreyvelich @thesuperzapper @franciscojavierarceo @varodrig @terrytangyuan

**Comment by Ashish241** [comment]
Add GitHub star count to homepage
Fixes #3624

Summary:
This PR adds a GitHub star count section to the homepage.

Details:
- Integrated star count display in hero section
- Matches existing CSS styling
- No additional heavy dependencies introduced
- Tested locally with Hugo v0.124.1

Screenshots:

<img width="1893" height="867" alt="Image" src="https://github.com/user-attachments/assets/9b7809c5-d046-4bb1-b1bc-2036c3be7aae" />
<img width="1893" height="870" alt="Image" src="https://github.com/user-attachments/assets/547a94a0-2d2d-4f31-8fb2-259a721cac8f" />

Let me know if any adjustments are required.

**Comment by Ashish241** [comment]
Hi team, this PR has been updated with the requested improvements and has been awaiting review for about a week. It was previously approved by one reviewer before being put on hold.

Could anyone take a look when you get a chance? Happy to make any additional changes if needed. Thanks!

Screenshot:

<img width="1906" height="872" alt="Image" src="https://github.com/user-attachments/assets/8e4cccfa-02dd-4c77-8250-52f0570aaed0" />
